contributors/devel/kubemark-guide.md

```diff
@@ -13,27 +13,35 @@ and how to use it.
 
 ## Architecture
 
-On a very high level Kubemark cluster consists of two parts: real master
-components and a set of “Hollow” Nodes. The prefix “Hollow” means an
-implementation/instantiation of a component with all “moving” parts mocked out.
+On a very high level, Kubemark cluster consists of two parts: a real master
+and a set of “Hollow” Nodes. From now on, the prefix “Hollow” to any component
+means an implementation/instantiation of the actual component with all “moving”
+parts mocked out. Essentially, a dummy component which just sends requests and
+receives responses but does no real work i.e. the internal logic is hollow.
 The best example is HollowKubelet, which pretends to be an ordinary Kubelet, but
 does not start anything, nor mount any volumes - it just lies it does. More
 detailed design and implementation details are at the end of this document.
 
-Currently master components run on a dedicated machine(s), and HollowNodes run
-on an ‘external’ Kubernetes cluster. This design has a slight advantage, over
-running master components on external cluster, of completely isolating master
-resources from everything else.
+Currently, master components (etcd, etcd-events, apiserver, controller-manager,
+scheduler) run on a dedicated machine as pods that are created/managed by kubelet,
+which itself runs as a systemd (or supervisord) service on the master VM.
+The HollowNodes on the other hand are run on an ‘external’ Kubernetes cluster.
+This design has a slight advantage, over running master components on an external
+cluster, of being able to completely isolate master resources from everything else.
 
 ## Requirements
 
 To run Kubemark you need a Kubernetes cluster (called `external cluster`)
 for running all your HollowNodes and a dedicated machine for a master.
-Master machine has to be directly routable from HollowNodes. You also need an
-access to some Docker repository.
+Master machine has to be directly routable from HollowNodes. You also need
+access to a Docker repository which has the container images for the
+master/HollowNode components (which is gcr.io in the case of GCE).
```

Comment:
I think this is not true for master components - we are just installing them in the correct paths and not pulling them from GCR (as they are not uploading every single build to gcr).